tests/contrib/celery/test_integration.py

```diff
@@ -31,90 +37,95 @@ def fn_task():
 
         traces = self.tracer.writer.pop_traces()
         eq_(1, len(traces))
-        eq_(2, len(traces[0]))
-        eq_('celery.apply', traces[0][0].name)
-        eq_('apply', traces[0][0].get_tag('celery.action'))
-        eq_('celery.run', traces[0][1].name)
-        eq_('run', traces[0][1].get_tag('celery.action'))
-        eq_('tests.contrib.celery.test_integration.fn_task', traces[0][0].resource)
-        eq_('tests.contrib.celery.test_integration.fn_task', traces[0][1].resource)
-        eq_('celery-producer', traces[0][0].service)
-        eq_('celery-worker', traces[0][1].service)
-        eq_('SUCCESS', traces[0][0].get_tag('state'))
-
-    def test_fn_task_bind(self):
-        # it should execute a traced task with a returning value
-        @self.app.task(bind=True)
-        def fn_task(self):
-            return self
+        eq_(1, len(traces[0]))
+
+    def test_idempotent_unpatch(self):
+        # calling unpatch() twice doesn't have side effects
+        unpatch()
+        unpatch()
+
+        @self.app.task
+        def fn_task():
+            return 42
 
         t = fn_task.apply()
         ok_(t.successful())
-        ok_('fn_task' in t.result.name)
+        eq_(42, t.result)
 
         traces = self.tracer.writer.pop_traces()
-        eq_(1, len(traces))
-        eq_(2, len(traces[0]))
-        eq_('celery.apply', traces[0][0].name)
-        eq_('apply', traces[0][0].get_tag('celery.action'))
-        eq_('celery.run', traces[0][1].name)
-        eq_('run', traces[0][1].get_tag('celery.action'))
-        eq_('tests.contrib.celery.test_integration.fn_task', traces[0][0].resource)
-        eq_('tests.contrib.celery.test_integration.fn_task', traces[0][1].resource)
-        eq_('celery-producer', traces[0][0].service)
-        eq_('celery-worker', traces[0][1].service)
-        eq_('SUCCESS', traces[0][0].get_tag('state'))
-
-    def test_fn_task_parameters(self):
-        # it should execute a traced task that has parameters
+        eq_(0, len(traces))
+
+    def test_fn_task_run(self):
+        # the body of the function is not instrumented so calling it
+        # directly doesn't create a trace
         @self.app.task
-        def fn_task_parameters(user, force_logout=False):
-            return (user, force_logout)
+        def fn_task():
+            return 42
+
+        t = fn_task.run()
+        eq_(t, 42)
+
+        traces = self.tracer.writer.pop_traces()
+        eq_(0, len(traces))
```

Comment:
This is a change in the test. Signals are not emitted for direct calls and `.run()` method. With the previous implementation this generated a Span.

Reply:
We should be sure to document this so that users are aware that `.run()` invocations will not generate spans.

Reply:
Made it clear in the documentation change. This is a kind of breaking change, but unfortunately we have a tradeoff here, between stability/safety and visibility.

cc @SeanOC